client/src/components/ActionMenu/ActionMenu.js

```diff
@@ -22,7 +22,15 @@ class ActionMenu extends PureComponent {
 
     // Force setting state to the end of the execution queue to clear a potential race condition
     // with entwine click handlers
-    window.setTimeout(() => this.setState((prevState) => ({ isOpen: !prevState.isOpen })), 0);
+    //
+    // eslint disable the next line which references this.state within setState()
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-access-state-in-setstate.md
+    // Following eslint guidelines this would be implemented as:
+    // window.setTimeout(() => this.setState((prevState) => ({ isOpen: !prevState.isOpen })), 0);
+    // However react ends up confused it's wrapped inside window.setTimeout()
+    // and it may end up setting state twice causing the toggle to open and then immediately close
+    // eslint-disable-next-line
```

Comment:
```suggestion
    // ignore linting rule because following the recommended replacement caused behat failures
    // eslint-disable-next-line no-access-state-in-setstate
```
Explicitly only disable the linting rule we want to disable.

We also don't need to go super in depth here, and in fact doing so just makes this comment distracting rather than helpful IMO.

Please make this change to the other changes as well.

Reply:
Updated